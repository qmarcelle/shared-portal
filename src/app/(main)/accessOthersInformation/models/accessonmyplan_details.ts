export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
