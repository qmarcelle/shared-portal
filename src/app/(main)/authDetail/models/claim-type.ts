export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
