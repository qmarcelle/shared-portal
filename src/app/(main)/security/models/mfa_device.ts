import { MfaDeviceType } from './mfa_device_type';

export interface MfaDevice {
  deviceType: MfaDeviceType;
  enabled: boolean;
  label: string;
  subLabel?: string;
  emailOrPhone: string | null;
  keyUri?: string;
  secret?: string;
}
