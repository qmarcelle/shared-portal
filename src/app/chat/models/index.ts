// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
