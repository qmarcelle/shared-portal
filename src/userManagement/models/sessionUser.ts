import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
