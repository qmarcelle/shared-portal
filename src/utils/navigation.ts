import { redirect } from 'next/navigation';

/**
 * Navigate to a new route programmatically outside of React components
 * @param path The path to navigate to
 */
export const navigateTo = (path: string) => {
  if (typeof window !== 'undefined') {
    window.location.href = path;
  } else {
    redirect(path);
  }
};

/**
 * Reload the current page
 */
export const reloadPage = () => {
  if (typeof window !== 'undefined') {
    window.location.reload();
  }
};

/**
 * Get the current URL
 * @returns The current URL or empty string if not in browser
 */
export const getCurrentUrl = () => {
  if (typeof window !== 'undefined') {
    return window.location.href;
  }
  return '';
};

/**
 * Get the current pathname
 * @returns The current pathname or empty string if not in browser
 */
export const getCurrentPathname = () => {
  if (typeof window !== 'undefined') {
    return window.location.pathname;
  }
  return '';
};

/**
 * Get URL search params
 * @returns URLSearchParams object or null if not in browser
 */
export const getSearchParams = () => {
  if (typeof window !== 'undefined') {
    return new URLSearchParams(window.location.search);
  }
  return null;
};
