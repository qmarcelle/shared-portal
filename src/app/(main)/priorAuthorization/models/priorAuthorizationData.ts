import { ClaimDetails } from '../../../../models/claim_details';

export const priorAuthorizationData: ClaimDetails[] = [
  {
    id: 'Claim1',
    claimStatus: 'Pending',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Physical Therapy',
    memberName: 'Chris Hall',
    serviceDate: '04/06/23',
    totalBilled: '',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Anand Patel',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim2',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Magnetic Resonance Imaging (MRI)',
    memberName: 'Chris Hall',
    serviceDate: '01/23/23',
    totalBilled: '145.00',
    planPaid: '114.76',
    myShare: '30.24',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim3',
    claimStatus: 'Partial Approval',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Continuous Positive Airway Pressure Machiâ€¦',
    memberName: 'Chris Hall',
    serviceDate: '08/10/23',
    totalBilled: '403.98',
    planPaid: '338.37',
    myShare: '65.61',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim4',
    claimStatus: 'Denied',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Home Health Services',
    memberName: 'Chris Hall',
    serviceDate: '07/05/23',
    totalBilled: '403.98',
    planPaid: '0.00',
    myShare: '403.98',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Home Health Services',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim5',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Brittant Johnson',
    memberName: 'Chris Hall',
    serviceDate: '07/04/23',
    totalBilled: '183.00',
    planPaid: '150.85',
    myShare: '32.15',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
];
