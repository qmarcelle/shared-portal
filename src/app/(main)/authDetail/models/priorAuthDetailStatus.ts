import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
