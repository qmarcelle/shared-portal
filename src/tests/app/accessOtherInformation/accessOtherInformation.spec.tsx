import AccessOthersInformation from '@/app/accessOthersInformation';
import AccessOthersInformationPage from '@/app/accessOthersInformation/page';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react';
import { redirect } from 'next/navigation';
// Mock the redirect function
jest.mock('next/navigation', () => ({
  redirect: jest.fn(),
}));

// Mock the auth function
jest.mock('src/auth', () => ({
  auth: jest.fn(),
}));

const renderUI = () => {
  return render(<AccessOthersInformation />);
};

describe('Communication accessOtherInformation', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should redirect to /dashboard if user role is PERSONAL_REP', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    mockAuth.mockResolvedValue({
      user: {
        currUsr: {
          role: UserRole.PERSONAL_REP,
        },
      },
    });
    await AccessOthersInformationPage();
    await waitFor(() => {
      expect(redirect).toHaveBeenCalledWith('/dashboard');
    });
  });

  it('should render accessOtherInformation if user role is not PERSONAL_REP', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    mockAuth.mockResolvedValue({
      user: {
        currUsr: {
          role: UserRole.MEMBER,
        },
      },
    });

    const component = renderUI();
    await waitFor(() => {
      expect(redirect).not.toHaveBeenCalled();
      expect(component).toMatchSnapshot();
    });
  });
});
