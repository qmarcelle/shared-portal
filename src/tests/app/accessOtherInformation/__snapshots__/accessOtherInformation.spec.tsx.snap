// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Communication accessOtherInformation should render accessOtherInformation if user role is not PERSONAL_REP 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="flex flex-col justify-center items-center page"
      >
        <div
          class="flex flex-col app-content app-base-font-color"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <h1
            class="title-1"
          >
            Access Others' Information
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <section
            class="flex flex-row items-start app-body"
          >
            <div
              class="flex flex-col flex-grow page-section-36_67 items-stretch"
            >
              <div
                class="card-main  large-section"
              >
                <div
                  class="flex flex-col items-stretch"
                >
                  <div>
                    <h1
                      class="title-1 title-2"
                    >
                      Understanding Access On My Plan
                    </h1>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
                    />
                  </div>
                  <div
                    class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                  >
                    <div
                      class="flex flex-row m-1 accordion-head"
                    >
                      <p
                        class="mr-2 flex-grow"
                      >
                        Access Granted
                      </p>
                      <img
                        alt="Down Chevron"
                        class="pl-2 w-6"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <div
                      class="mr-2 flex-grow"
                    />
                  </div>
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                  <div
                    class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                  >
                    <div
                      class="flex flex-row m-1 accordion-head"
                    >
                      <p
                        class="mr-2 flex-grow"
                      >
                        Request Access
                      </p>
                      <img
                        alt="Down Chevron"
                        class="pl-2 w-6"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <div
                      class="mr-2 flex-grow"
                    />
                  </div>
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                  <div
                    class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                  >
                    <div
                      class="flex flex-row m-1 accordion-head"
                    >
                      <p
                        class="mr-2 flex-grow"
                      >
                        Special Permissions
                      </p>
                      <img
                        alt="Down Chevron"
                        class="pl-2 w-6"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <div
                      class="mr-2 flex-grow"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col page-section-63_33 items-stretch"
            >
              <div
                class="card-main large-section"
              >
                <div
                  class="flex flex-col flex flex-col"
                >
                  <div
                    class="flex flex-col"
                  >
                    <h2
                      class="title-2"
                    >
                      On My Plan
                    </h2>
                  </div>
                  <div>
                     
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1"
                        style="display: block;"
                      >
                        Below is the access you've been granted by other members on your plan.
                      </p>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                    />
                  </div>
                  <div
                    class="flex flex-col flex flex-col"
                  />
                </div>
              </div>
            </div>
          </section>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color"
      >
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <h1
          class="title-1"
        >
          Access Others' Information
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main  large-section"
            >
              <div
                class="flex flex-col items-stretch"
              >
                <div>
                  <h1
                    class="title-1 title-2"
                  >
                    Understanding Access On My Plan
                  </h1>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
                  />
                </div>
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      Access Granted
                    </p>
                    <img
                      alt="Down Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="mr-2 flex-grow"
                  />
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      Request Access
                    </p>
                    <img
                      alt="Down Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="mr-2 flex-grow"
                  />
                </div>
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <div
                  class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
                >
                  <div
                    class="flex flex-row m-1 accordion-head"
                  >
                    <p
                      class="mr-2 flex-grow"
                    >
                      Special Permissions
                    </p>
                    <img
                      alt="Down Chevron"
                      class="pl-2 w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <div
                    class="mr-2 flex-grow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col page-section-63_33 items-stretch"
          >
            <div
              class="card-main large-section"
            >
              <div
                class="flex flex-col flex flex-col"
              >
                <div
                  class="flex flex-col"
                >
                  <h2
                    class="title-2"
                  >
                    On My Plan
                  </h2>
                </div>
                <div>
                   
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                  <div
                    class="flex flex-col"
                  >
                    <p
                      class="body-1"
                      style="display: block;"
                    >
                      Below is the access you've been granted by other members on your plan.
                    </p>
                  </div>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                  />
                </div>
                <div
                  class="flex flex-col flex flex-col"
                />
              </div>
            </div>
          </div>
        </section>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
